trie-db/src/lib.rs

```diff
@@ -530,21 +363,29 @@ pub trait TrieLayout {
 	type Hash: Hasher;
 	/// Codec to use (needs to match hasher and nibble ops).
 	type Codec: NodeCodec<HashOut = <Self::Hash as Hasher>::Out>;
+	type Location: Location;
```

Comment:
`const USE_EXTENSION` and all extension related code looks easy enough to remove to me, but can be the first pr of the new code base.